### PR TITLE
Updgrade to whisper 3

### PR DESCRIPTION
**Pull request by KillerX**
This is essentially a rewrite because I have now switched to the more generic transformers library that allows us to easily build pipelines for other models as well should we want it, and whisper-3 anyways had some issues with the old approach.

Note: The code is tested on the machine it will run on. I don't guarantee that all the GPU stuff and requirements are correct for it to run on a mac.

### CODE REVIEW

File: transcription-handler.go [hunk 1]
```diff
@@ -66,7 +66,7 @@ func (h *handlers) SubmitJob(c *gin.Context) {
 	}
 
 	if job.Model == "" {
-		job.Model = "large-v3"
+		job.Model = "large-v2"
```
Comment (sifferhans) [inline transcription-handler.go]: Why was this downgraded?

**Review by sifferhans — APPROVED** [review]
Looks good to me. I just skimmed through though

File: transcription-handler.go (hunk 1 above)
Comment (KillerX) [reply transcription-handler.go]: I was testing something, but it appears that it is not used at all. Thanks for spotting it'll fix it up.

Comment (KillerX) [reply transcription-handler.go]: I actually respect the value passed in now but filter it on known good values. Also updated all defaults to V3

**Pull request merged**